src/vcSHP.cpp

```diff
@@ -0,0 +1,741 @@
+      readPosition = UnpackDouble(pSHP->localBigEndian, readPosition, &record.data.multiPointM.MRange.y, buffer);
+
+      record.data.multiPointM.MValue = udAllocType(double, record.data.multiPointM.pointsCount, udAF_Zero);
+      for (int i = 0; i < record.data.multiPointM.pointsCount; i++)
+        readPosition = UnpackDouble(pSHP->localBigEndian, readPosition, &record.data.multiPointM.MValue[i], buffer);
+    }
+  }
+  break;
+  default:
+  break;
+  }
+
+  pSHP->shpRecords.PushBack(record);
+  offset += readPosition;
+
+epilogue:
+  if (buffer)
+    udFree(buffer);
+
+  if (result != udR_Success)
+    vcSHP_ReleaseRecord(record);
+
+  return result;
+}
+
+udResult vcSHP_LoadShpFile(vcSHP *pSHP, const char *pFilename)
+{
+  udResult result;
+
+  udFile *pFile = nullptr;
+  uint8_t *headerBuffer = nullptr;
+  uint32_t offset = 0;
+
+  vcSHP_Header &header = pSHP->shpHeader;
+  UD_ERROR_CHECK(udFile_Open(&pFile, pFilename, udFOF_Read));
+
+  headerBuffer = udAllocType(uint8_t, 100, udAF_Zero);
+  UD_ERROR_CHECK(udFile_Read(pFile, headerBuffer, 100 * sizeof(uint8_t)));
+  // 00 00 27 0a
+  UD_ERROR_IF((headerBuffer[0] != 0 || headerBuffer[1] != 0 || headerBuffer[2] != 0x27 || headerBuffer[3] != 0x0a), udR_ParseError);
+
+  header.fileLength = (headerBuffer[24] << 24) | (headerBuffer[25] << 16) | (headerBuffer[26] << 8) | headerBuffer[27];
+  if (header.fileLength < UINT_MAX / 2)
+    header.fileLength *= 2;
+  else
+    header.fileLength = (UINT_MAX / 2) * 2;
+
+  header.version = (headerBuffer[31] << 24) | (headerBuffer[30] << 16) | (headerBuffer[29] << 8) | headerBuffer[28];
+  header.shapeType = (vcSHP_Type)((int32_t)((headerBuffer[35] << 24) | (headerBuffer[34] << 16) | (headerBuffer[33] << 8) | headerBuffer[32]));
+
+  offset = 36;
+  offset = UnpackDouble(pSHP->localBigEndian, offset, &header.MBRmin.x, headerBuffer);
+  offset = UnpackDouble(pSHP->localBigEndian, offset, &header.MBRmin.y, headerBuffer);
+  offset = UnpackDouble(pSHP->localBigEndian, offset, &header.MBRmax.x, headerBuffer);
+  offset = UnpackDouble(pSHP->localBigEndian, offset, &header.MBRmax.y, headerBuffer);
+  offset = UnpackDouble(pSHP->localBigEndian, offset, &header.ZRange.x, headerBuffer);
+  offset = UnpackDouble(pSHP->localBigEndian, offset, &header.ZRange.y, headerBuffer);
+  offset = UnpackDouble(pSHP->localBigEndian, offset, &header.MRange.x, headerBuffer);
+  offset = UnpackDouble(pSHP->localBigEndian, offset, &header.MRange.y, headerBuffer);
+
+  UD_ERROR_IF(offset != 100, udR_ParseError);
+  udFree(headerBuffer);
+
+  UD_ERROR_CHECK(pSHP->shpRecords.Init(32));
+  while (offset < header.fileLength)
+  {
+    result = vcSHP_LoadShpRecord(pSHP, pFile, offset, header.shapeType);
+    UD_ERROR_CHECK(result);
+  }
+
+epilogue:
+  udFile_Close(&pFile);
+
+  if (headerBuffer)
+    udFree(headerBuffer);
+
+  return result;
+}
+
+void vcSHP_LoadDbfFile(vcSHP *pSHP, const char *pFilename)
```

Comment:
This function isn't necessary, you only need to call `vcDBF_Load` as it will allocate the `vcDBF` object. `vcDBF_Create` is intended for when you want to make your own DBF file, rather than use an existing one.

Reply:
fixed.